src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -1450,8 +1450,20 @@ MulticopterPositionControl::control_non_manual()
 	/* use constant descend rate when landing, ignore altitude setpoint */
 	if (_pos_sp_triplet.current.valid
 	    && _pos_sp_triplet.current.type == position_setpoint_s::SETPOINT_TYPE_LAND) {
-		_vel_sp(2) = _land_speed.get();
 		_run_alt_control = false;
+
```

Comment:
This whole block is duplicating the existing functionality: https://github.com/PX4/Firmware/blob/master/src/modules/mc_pos_control/mc_pos_control_main.cpp#L2439-L2443
Did it not work correctly for you?

Reply:
Look at the one liner that I removed.

 _vel_sp(2) = _land_speed.get();

Clearly this is overriding the current Z vel setpoint when the setpoint type is LAND.

Also, always limiting veritial velocity based upon height does not seem right. Those parameters are specified with the word 'LAND' and thus should only affect flight characteristics while in the LAND mode (this is my opinion).